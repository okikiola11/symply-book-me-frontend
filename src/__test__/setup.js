/* eslint-disable import/no-extraneous-dependencies */
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { fireEvent, render } from '@testing-library/react';
import thunk from 'redux-thunk';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { mount, shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';

Enzyme.configure({ adapter: new Adapter() });

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const store = mockStore({
  auth: {
    user: {
      appointment: [],
      id: 'id-01',
      email: 'test@mail.com',
    },
  },
  isLoading: false,
  error: null,
});

export const shallowRender = (Element, props = {}) => (
  shallow(
    <Provider store={store}>
      <BrowserRouter>
        <Element {...props} />
      </BrowserRouter>
    </Provider>,
  )
);

export const shallowRenderWithoutProvider = (Element, props = {}) => (
  shallow(
    <Element {...props} />,
  )
);

export const mountRender = (Element, props = {}) => (
  mount(
    <Provider store={store}>
      <BrowserRouter>
        <Element {...props} />
      </BrowserRouter>
    </Provider>,
  )
);

export const renderWrapper = (Element, props = {}) => {
  const { getByLabelText, container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Element {...props} />
      </BrowserRouter>
    </Provider>,
  );
  return { getByLabelText, container };
};

export const changeInputText = async (element, value = '') => {
  await act(async () => {
    fireEvent.change(element, { target: { value } });
    fireEvent.blur(element);
  });
};

export const getById = (container, id) => container.querySelector(`#${id}`);
