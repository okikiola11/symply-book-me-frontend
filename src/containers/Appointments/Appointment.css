.content {
  padding-left: 15%;
}
