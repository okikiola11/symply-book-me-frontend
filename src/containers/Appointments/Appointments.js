import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import Appointment from '../../components/Appointment/Appointment';
import Sidebar from '../Sidebar/Sidebar';
import { fetchAppointments, deleteAppointments } from '../../api';
import { getAppointments, destroyAppointments } from '../../Redux/actions';

const Appointments = () => {
  const dispatch = useDispatch();
  const appointments = useSelector((state) => state.appointments);
  // const [isLoaded, setIsLoaded] = useState(null);

  useEffect(() => {
    const getAllAppointments = async () => {
      if (localStorage.getItem('token')) {
        const value = await fetchAppointments();
        dispatch(getAppointments(value));
      }
    };
    getAllAppointments();
    // if (appointments?.length > 0) {
    //   setIsLoaded(true);
    // }
    // return () => {
    //   setIsLoaded(false);
    // };
  }, []);

  const handleDeleteAppointment = async (e) => {
    const { id } = e.target.dataset;
    const action = await deleteAppointments(id);
    dispatch(destroyAppointments(action));
  };

  return (
    <>
      <Sidebar />
      {appointments?.length ? (
        <section className="content">
          <h2 className="font-weight-bold text-center pb-5">APPOINTMENTS</h2>
          <div className="table-responsive">
            <table className="container table table-striped">
              <thead>
                <tr>
                  <th>Lawyer&apos;s Name</th>
                  <th>Email</th>
                  <th>Date</th>
                  <th>City</th>
                  <th>Cancel Appointment</th>
                </tr>
              </thead>
              <tbody>
                {appointments.map((appointment) => (
                  <Appointment
                    key={appointment.id}
                    appointment={appointment}
                    handleClick={handleDeleteAppointment}
                    dataId={appointment?.id}
                  />
                ))}
              </tbody>
            </table>
          </div>
        </section>
      ) : (
        <section className="content text-center">
          <h4 className="py-5 font-weight-bold  mt-5">No appointments yet!</h4>
        </section>
      )}
    </>
  );
};

export default Appointments;
