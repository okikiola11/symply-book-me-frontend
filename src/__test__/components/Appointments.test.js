import { shallowRender } from '../setup';
import Appointments from '../../components/Appointment/Appointment';

// describe('Appointments', () => {
//   const page = shallowRender(Appointments);

//   test('Should render Appointments', () => {
//     expect(page).toBeTruthy();
//     expect(page).not.toBeFalsy();
//     expect(page).toMatchSnapshot();
//   });
// });