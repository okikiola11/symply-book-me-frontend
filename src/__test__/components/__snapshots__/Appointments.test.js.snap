// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointments Should render Appointments 1`] = `ShallowWrapper {}`;
